docs/protocol.md

```diff
@@ -0,0 +1,333 @@
+# Introduction
+
+This document describes version 7 of the Raw rai protocol (as mentioned in https://github.com/nanocurrency/raiblocks/wiki/Network-usage).
+
+**Note that this document is a work in progress and is not complete and/or fully correct yet!**
+
+Code references used:
+- `rai/node/common.cpp`
+  - `rai::message`
+  - `rai::message::read_header()`
+  - `rai::message::block_type_set()`
+- `rai/node/common.hpp`
+  - `enum class message_type`
+- `rai/lib/blocks.hpp`
+  - `enum class block_type`
+
+## Values
+
+The following different types of values are used in the protocol:
+
+  * Account
+    - The 256-bit Ed25519 public key of an account
+    - Encoded as a sequence of 32 bytes
+    - Note that the well-known `xrb_...` notation for an address/representative
+      is a different encoding of this value
+  * Representative
+    - Same as Account, but in a different role
+  * Block hash
+    - The 256-bit hash of a block  
+    - The hash is computed from a set of "hashables": fields in a block that are hashed
+      with the Blake2b hashing function
+    - Unless noted otherwise the hashables for a block are the block-specific fields
+      listed in the table with each block description. They are to be hashed in the
+      order they are listed.
+    - The block hash is used (amongst other things) to make references between blocks
+    - Encoded as a sequence of 32 bytes
+  * Balance / Amount
+    - A 128-bit unsigned integer
+    - This number has unit "raw", where 1 XRB = 1 Mxrb = 10<sup>30</sup> raw.
+      See [here](https://github.com/nanocurrency/raiblocks/wiki/Distribution,-Mining-and-Units)
+      for more information.
+    - Encoded as a sequence of 16 bytes, *big-endian* order
+  * Signature
+      - A 512-bit Ed25519 signature
+      - Encoded as a sequence of 64 bytes
+  * Work
```

Comment:
You could add that the work is normally computed based upon the previous block's hash, except on open blocks where the new address's public key is used instead (should apply to state blocks accordingly) AFAIK. 

Another interesting detail may be the difference in the work thresholds for the different networks.